sympy/calculus/util.py

```diff
@@ -480,7 +480,7 @@ def _check(orig_f, period):
 
     elif f.is_Mul:
         coeff, g = f.as_independent(symbol, as_Add=False)
-        if isinstance(g, TrigonometricFunction) or coeff is not S.One:
+        if isinstance(g, TrigonometricFunction) or coeff is not S.One and not (coeff.is_Float and coeff == 1.0):
```

Comment:
We should really make a function for testing this sort of thing.

Looking at `git grep 'is S.One'` I see a lot of other places in the codebase that potentially have the same problem.

Reply:
I think it better to use `coeff != 1` here.

Reply:
Using `!=` does work but I would rather have a function specifically for testing integers and integer-valued floats (see #20033)

Reply:
A bit off-topic and I believe people have talked it a lot, but I don't understand why Sym(bolic)Py should support calculations mixing numeric and symbolic.

Reply:
In practice an analysis-oriented CAS that doesn't support approximate numerics is not very useful because most real problems cannot be solved symbolically.

Reply:
Yes the `'is not S.One'` line is a part of multiple `if` routines in the code. Should I make a private function and check everytime ?

Reply:
Also kindly look into the `CircleCI` test fail .All other tests have passed .

Reply:
> Also kindly look into the `CircleCI` test fail .All other tests have passed .

You have to rebase your code with latest master, so that latest commits into the master are available on your local machine. You can do that by:
```
git checkout master
git fetch upstream
git merge upstream/master
git checkout fix_23041
git rebase master
```

Reply:
The CircleCI check doesn't matter so nothing needs to be done about that.

I think maybe what is needed is a function that can check for when two expressions are equal except for possible differences where one expression has a float and the other has an int. For this particular case though to fix this particular issue we could just use `coeff != 1` for now to keep it simple.

Reply:
Thanks @eagleoflqj for the commits.I have been busy with travel ,hence could not respond to the desired suggestions. Thanks again !